### PR TITLE
Volunteer-team-preference

### PR DESCRIPTION
**Pull request by H-BAHDON**
This is a:

- [x] ✨ **New feature** - new behaviour has been implemented
- [ ] 🐛 **Bug fix** - existing behaviour has been made to behave
- [ ] ♻️ **Refactor** - the behaviour has not changed, just the implementation
- [ ] ✅ **Test backfill** - tests for existing behaviour were added but the behaviour itself hasn't changed
- [ ] ⚙️ **Chore** - maintenance task, behaviour and implementation haven't changed

### Description

- **Purpose** - 
- The purpose of this ticket is to enable CYF Staff members to streamline the onboarding process by requesting volunteers to identify the specific team they would like to volunteer for during the sign-up process, this will accelerating the overall onboarding procedure.

- **How to check** - 

- Verify that the list is displaying all the teams {Education, Mentorship, Outreach, Personal Development, Tech Project} in alphabetical order.

- Create a new team in the MongoDB under the "teams" section. You can accomplish this by cloning one of the existing teams and changing its name to "New Team."

- Check the form to ensure that the newly created team ("New Team") is added as an option.

- Navigate to the dashboard at /volunteers and confirm that the "New Team" is correctly added to the teams list.

- Select one user from the dashboard and modify their team assignment, changing it to the newly created "New Team."

- Delete the "New Team" you have created from the MongoDB.

- After deleting the "New Team," verify that the dashboard for the selected user automatically updates to display "Select Team" as the option.

### Author checklist

- [x] I have written a title that reflects the relevant ticket
- [x] I have written a description that says what the PR does and how to validate it
- [x] I have linked to the project board ticket (and any related PRs/issues) in the Links section
- [x] I have added a link to this PR to the ticket
- [x] I have made the PR to `qa` from a branch named `<category>/<name>`, e.g. `feature/edit-spaceships` or `bugfix/restore-oxygen`
- [ ] I have completed the manual tests [described here](https://github.com/CodeYourFuture/tech-team/wiki/3.-How-we-work#manual-test-procedures)
- [ ] I have requested reviewers here and in my team chat channel

- [ ] I have spoken with my PM or TL about any parts of this task that may have become out-of-scope, or any additional improvements that I now realise may benefit my project
- [ ] I have added tests, or new tests were not required
- [ ] I have updated any documentation (e.g. diagrams, schemas), or documentation updates were not required

### CODE REVIEW

**Review by SaadiaELF — APPROVED** [review]
LGFM, all tests passed locally , and meet the scenarios on the ticket.
Well done @H-BAHDON , I'm really impressed by your perseverance !

File: src/Components/forms/inputs/index.js [hunk 1]
```diff
@@ -104,6 +106,28 @@ export default class VolunteerForm extends Component {
           placeholder="Just one or two sentences as required..."
           type="textarea"
         />
+        <DropDown
+          data-testid="testId"
```
[the remark this answers is not in the document]
Comment (H-BAHDON) [reply src/Components/forms/inputs/index.js]: [Removed data-testid=testId](https://github.com/CodeYourFuture/cyf-forms/pull/133/commits/8826dd1bfc0580c778e5189cd676b2b10929f535)

File: src/Components/forms/index.js [hunk 2]
```diff
@@ -11,7 +11,27 @@ import './index.css'
...
+
+  fetchTeamData = async () => {
+    try {
+      const response = await axios.get(`${domain()}/teams`)
+      const teamData = response.data.teams
+      teamData.sort((a, b) => a.name.localeCompare(b.name))
+
+      this.setState({ teamOptions: teamData })
+    } catch (error) {
+      console.error('Error fetching team data:', error)
```
Comment (CanaryWharf) [inline src/Components/forms/index.js:32]: Should we have a way of displaying the errors to the user? The user may not know to check for console errors if things freeze up or don't display, and I feel like we should be providing feedback to the user if something goes wrong.

Comment (H-BAHDON) [reply src/Components/forms/index.js]: I Updated teamFetchData: Reusing the error message from handleMagicLinkRequest function,
```
 return this.setState({
          err: 'Sorry, we are currently experiencing technical issues, please try again later.',
          loading: false
        })
```
 and tested this but closing the api and only running the CYF-Form, as the API is not running anymore, i got the msg
 
 
image

**Review by CanaryWharf — APPROVED** [review]
Different teams load and appear as options. LGTM

**Pull request merged**